gui/src/lib/gui-theme.ts

```diff
@@ -0,0 +1,34 @@
+import { writable } from 'svelte/store';
+
+export const guiThemeLocalStorageKey = 'io.deref.exo/gui-theme';
+
+function createGuiTheme() {
+  const ls = localStorage.getItem(guiThemeLocalStorageKey) || 'auto';
+
+  const { subscribe, set } = writable(ls);
+
+  const localSyncedSet = (s: string) => {
+    set(s);
+    localStorage.setItem(guiThemeLocalStorageKey, s);
+  };
+
+  return {
+    subscribe,
+    auto: () => localSyncedSet('auto'),
```

Comment:
why are there 4 methods here rather than `set(value: ThemeName)` or something like that?